pkg/server/commands/connectedobjects/connected_objects.go

```diff
@@ -446,6 +462,7 @@ func (c *ConnectedObjectsQuery) reverseExpandDirect(
 		return err
 	}
 	defer iter.Stop()
+	atomic.AddUint32(resolutionMetadata.QueryCount, 1)
```

Comment:
😢 this adds a lot of concurrency contention.

Reply:
Why a lot? It's not like trying to send through a channel that might be full 🤔